### PR TITLE
[feature_file] Create feature process file

### PR DESCRIPTION
**Pull request by jasondlee**
As discussed in process and tracking tool discussion, port Brian's table to adoc format in this repo

### CODE REVIEW

**Comment by jmesnil** [comment]
@jasondlee I've opened https://github.com/jasondlee/wildfly-proposals/pull/1 to improve the table readability. I did not heavily touched the content though

File: FEATURE_PROCESS.adoc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+|Stable API and behavior
+
+All hard requirements in analysis covered
+|Domain Transformation
+|Encouraged
+|Encouraged
+|Encouraged
+|Required
+|Component Validation
+|Acceptable Open Source License
```
Comment (jmesnil) [inline FEATURE_PROCESS.adoc:67]: do we have somewhere a list of such license?

File: FEATURE_PROCESS.adoc [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+|Experimental plus:
+
+Uses maintained components
+
+Component available from JBoss Nexus or maven central
```
Comment (jmesnil) [inline FEATURE_PROCESS.adoc:72]: Java components available from Maven repositories (JBoss Nexus or Maven Central)

File: FEATURE_PROCESS.adoc (hunk 1 above)
Comment (jasondlee) [reply FEATURE_PROCESS.adoc:67]: Not that I'm aware of. Perhaps @bstansberry knows?

File: FEATURE_PROCESS.adoc (hunk 2 above)
Comment (jasondlee) [reply FEATURE_PROCESS.adoc]: Updated.

**Review by jmesnil — APPROVED** [review]

**Comment by jmesnil** [comment]
@jasondlee thanks!

**Pull request merged**